src/server/middleware/security/auth_proxy.go

```diff
@@ -0,0 +1,104 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package security
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/goharbor/harbor/src/common"
+	"github.com/goharbor/harbor/src/common/dao"
+	"github.com/goharbor/harbor/src/common/models"
+	"github.com/goharbor/harbor/src/common/security"
+	"github.com/goharbor/harbor/src/common/security/local"
+	"github.com/goharbor/harbor/src/common/utils/log"
+	"github.com/goharbor/harbor/src/core/auth"
+	"github.com/goharbor/harbor/src/core/config"
+	"github.com/goharbor/harbor/src/internal"
+	"github.com/goharbor/harbor/src/pkg/authproxy"
+)
+
+type authProxy struct{}
+
+func (a *authProxy) Generate(req *http.Request) security.Context {
+	log := log.G(req.Context())
+	if internal.GetAuthMode(req.Context()) != common.HTTPAuth {
+		return nil
+	}
+	// only support docker login
+	if req.URL.Path != "/service/token" {
```

Comment:
Oh this should be updated in the `v2` -> basic auth refactor , I think it should check if it has /v2 prefix